packages/grid/x-data-grid-premium/src/components/GridColumnMenuRowGroupItem.tsx

```diff
@@ -30,9 +30,10 @@ function GridColumnMenuRowGroupItem(props: GridColumnMenuItemProps) {
       onClick(event);
     };
 
-    const name = columnsLookup[field].headerName ?? field;
+    const groupedColumn = columnsLookup[field];
+    const name = groupedColumn.headerName ?? field;
     return (
-      <MenuItem onClick={ungroupColumn} key={field}>
+      <MenuItem onClick={ungroupColumn} key={field} disabled={!groupedColumn.groupable}>
```

Comment:
I made an exception here as compared to the other `colDef` flags, i.e the item will be shown in the column menu for the grouping column in the disabled state for grouped columns that have `colDef.groupable = false`.

In this example company's `colDef` has `groupable = false`
![image](https://github.com/mui/mui-x/assets/12609561/1b7c2ab4-b60b-473f-bb97-cd3b399147cd)

The rationale is, that if a grouping is applied on a non-groupable column, there's no way for the users to check which column is grouped as the header says `Group`, so keep showing the disabled (read-only) column menu item will be a pointer that this specific column is grouped.

Reply:
Makes sense to me